### PR TITLE
Bump version to 0.23.0

### PR DESCRIPTION
**Pull request by tgaddair**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,13 +8,43 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+
+### Deprecated
+
+### Removed
+
+### Fixed
+
+## [v0.23.0] - 2021-10-06
+
+### Added
```
Comment (chongxiaoc) [inline CHANGELOG.md:21]: Where is this change?
```
### Removed
Spark/Keras: remove bare Keras support. (#3191)
```

Comment (tgaddair) [reply CHANGELOG.md:21]: It was already added in a previous commit, line 53.

**Comment by tgaddair** [comment]
> Also, I have a question about `tensorflow==2.6.0`. We haven't fixed `import keras` usage for `tensorflow` from #3157
> 
> But our CI tests with `tf_2_6_0` don't fail at this point?

I think it still technically works for now, but is officially unsupported. @DEKHTIARJonathan may know more.

**Review by chongxiaoc — APPROVED** [review]

**Review by EnricoMi — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -25,8 +55,24 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
...
+
+- Fixed bug in Lightning Profiler on Ray. ([#3122](https://github.com/horovod/horovod/pull/3122))
+
+- Fixed torch op lazy release to prevent OOM in elastic training. ([#3110](https://github.com/horovod/horovod/pull/3110))
+
+- Lightning: Fixed usage of the checkpoint callback. ([#3186](https://github.com/horovod/horovod/pull/3186))
+
+- Fixed MPICH support to use Intel MPI's implementation. ([#3148](https://github.com/horovod/horovod/pull/3148))
+
+- Fixed CMake builds to be compatible with older versions of CMake. ([#3196](https://github.com/horovod/horovod/pull/3196))
```
[the remark this answers is not in the document]
Comment (tgaddair) [reply CHANGELOG.md:73]: Makes sense, thanks.

**Pull request merged**